### PR TITLE
Updating heapster version to v0.14.1.

### PR DESCRIPTION
**Pull request by vishh**
Adding a standalone version of heapster which exposes stats via REST API.

### CODE REVIEW

**Comment by vishh** [comment]
cc @a-robinson

**Comment by vishh** [comment]
cc @saad-ali for the salt changes.

File: cluster/addons/cluster-monitoring/google/heapster-controller.yaml [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
+        version: v3
         kubernetes.io/cluster-service: "true"
     spec: 
       containers: 
-        - image: gcr.io/google_containers/heapster:v0.14.0
+        - image: gcr.io/google_containers/heapster:v0.14.1
           name: heapster
           command:
             - /heapster
-            - --source=kubernetes:https://kubernetes
```
Comment (saad-ali) [inline cluster/addons/cluster-monitoring/google/heapster-controller.yaml]: The source is no longer required?

Comment (vishh) [inline cluster/addons/cluster-monitoring/google/heapster-controller.yaml]: API server url isn't required. Heapster will pick up from the configs in the token.

Comment (saad-ali) [inline cluster/addons/cluster-monitoring/google/heapster-controller.yaml]: Ok

**Comment by saad-ali** [comment]
LGTM. Will let @a-robinson ok from GKE side.

File: cluster/gke/config-default.sh [hunk 2]
```diff
@@ -34,4 +34,5 @@ ELASTICSEARCH_LOGGING_REPLICAS=1
 #   none     - No cluster monitoring setup 
 #   influxdb - Heapster, InfluxDB, and Grafana 
 #   google   - Heapster, Google Cloud Monitoring, and Google Cloud Logging
-ENABLE_CLUSTER_MONITORING="${KUBE_ENABLE_CLUSTER_MONITORING:-none}"
+#   standalone     - Heapster only. Metrics available via Heapster REST API.
+ENABLE_CLUSTER_MONITORING="${KUBE_ENABLE_CLUSTER_MONITORING:-standalone}"
```
Comment (a-robinson) [inline cluster/gke/config-default.sh]: For what it's worth, this env var doesn't actually control anything

**Comment by a-robinson** [comment]
LGTM

File: cluster/gke/config-default.sh (hunk 2 above)
Comment (saad-ali) [inline cluster/gke/config-default.sh]: Worth removing?

Comment (vishh) [inline cluster/gke/config-default.sh]: So is this entire file obsolete?

**Comment by vishh** [comment]
@saad-ali: This PR needs to be merged soon since it needs to be included in v0.19.1. Thanks!

File: cluster/gke/config-default.sh (hunk 2 above)
Comment (vishh) [inline cluster/gke/config-default.sh]: We can do it in a separate PR.

Comment (a-robinson) [inline cluster/gke/config-default.sh]: We just need to scrub the options that won't do anything. It's been on my backburner for a little while, but hasn't been all that important. Don't worry about it in this pr.
